components/tabs/src/tabs.udl

```diff
@@ -51,9 +56,37 @@ dictionary ClientRemoteTabs {
     sequence<RemoteTabRecord> remote_tabs;
 };
 
-dictionary TabsRequestedClose {
-    string client_id;
-    sequence<string> urls;
+[Enum]
+interface RemoteCommand {
+    CloseTab(string url);
+    // CloseInactive(); ??
+};
+
+interface RemoteCommandStore {
+    // Add a new command, after which it will be pending. Returns false if the command is already active.
+    [Throws=TabsApiError]
+    boolean add_remote_command([ByRef] string device_id, [ByRef] RemoteCommand command);
+
+    // Add a new command with an explicit timestamp. Primarily used by tests.
+    [Throws=TabsApiError]
+    boolean add_remote_command_at([ByRef] string device_id, [ByRef] RemoteCommand command, Timestamp when);
+
+    [Throws=TabsApiError]
+    boolean remove_remote_command([ByRef] string device_id, [ByRef] RemoteCommand command);
+
+    // This is for the code sending the commands, result is sorted by time_requested.
+    [Throws=TabsApiError]
+    sequence<PendingCommand> get_unsent_commands();
+
+    [Throws=TabsApiError]
+    boolean set_remote_command_sent([ByRef] string device_id, [ByRef] RemoteCommand command);
```

Comment:
Design nit / question: I wonder if it makes sense to have `set_remote_command_sent` take a `PendingCommand`, instead of `(device_id, RemoteCommand)`—to make the expected flow of these commands a little clearer.

Reply:
Yeah, that does make sense - but doing that means I hated the old name, so we now have `set_pending_command_sent`